src/compiler/emitter.ts

```diff
@@ -2630,15 +2630,21 @@ module ts {
             }
 
             function emitSuper(node: Node) {
-                var flags = resolver.getNodeCheckFlags(node);
-                if (flags & NodeCheckFlags.SuperInstance) {
-                    write("_super.prototype");
-                }
-                else if (flags & NodeCheckFlags.SuperStatic) {
-                    write("_super");
+                if (languageVersion >= ScriptTarget.ES6) {
+                    write("super");
                 }
                 else {
-                    write("super");
+                    Debug.assert(languageVersion < ScriptTarget.ES6)

```

Comment:
not sure you really need that.. it is the else of the if statement with the same condition. just looks like duplication to me,